tools/deprecated-config-checker/deprecated-config.yaml

```diff
@@ -0,0 +1,70 @@
+
+schema_config:
+  configs:
+    store:
+      _deprecated: ["aws", "aws-dynamo", "gcp", "gcp-columnkey", "bigtable", "bigtable-hashed", "cassandra", "grpc"]
+      _msg: "Use tsdb (preferred) or boltdb-shipper instead."
+    object_store:
+      _deprecated: ["aws-dynamo", "gcp", "gcp-columnkey", "bigtable", "bigtable-hashed", "cassandra", "grpc"]
+      _msg: "Use any other non-deprecated storage instead."
+
+storage_config:
+  bigtable: "Bigtable is deprecated. Use any other non-deprecated storage instead."
+  cassandra: "Cassandra is deprecated. Use any other non-deprecated storage instead."
+  boltdb: "Boltdb is deprecated. Consider using boltdb_shipper or any other non-deprecated storage instead."
+  grpc_store: "gRPC store is deprecated. Use any other non-deprecated storage instead."
+  aws:
+    dynamodb: "DynamoDB is deprecated and will be eventually removed"
+
+chunk_store_config:
+  write_dedupe_cache_config: "Write dedupe cache is deprecated along with deprecated index types. Consider using TSDB index which does not require a write dedupe cache."
+  max_look_back_period: "Use global or per-tenant max_query_lookback instead."
+
+## NOTE: This will also be used to validate per-tenant overrides.
+limits_config:
+  unordered_writes: "Will be eventually removed."
+  ruler_evaluation_delay_duration: "Will be eventually removed."
+  ruler_remote_write_url: "Use ruler_remote_write_config instead."
+  ruler_remote_write_timeout: "Use ruler_remote_write_config instead."
+  ruler_remote_write_headers: "Use ruler_remote_write_config instead."
+  ruler_remote_write_relabel_configs: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_capacity: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_min_shards: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_max_shards: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_max_samples_per_send: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_batch_send_deadline: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_min_backoff: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_max_backoff: "Use ruler_remote_write_config instead."
+  ruler_remote_write_queue_retry_on_ratelimit: "Use ruler_remote_write_config instead."
+  ruler_remote_write_sigv4_config: "Use ruler_remote_write_config instead."
+  per_tenant_override_config: "Feature renamed to 'runtime configuration', flag deprecated in favor of runtime_config.file"
+  per_tenant_override_period: "Feature renamed to 'runtime configuration', flag deprecated in favor of runtime_config.period"
+  allow_deletes: "Use deletion_mode per tenant configuration instead."
+
+# TODO:
+# - ruler.storage: Use -ruler-storage.
+#      I cannot find where ruler-storage is defined.
+# - index_gateway.replication_factor: Use -index-gateway.shard-size instead. The shard size is also a per-tenant setting.
```

Comment:
this one I think https://github.com/grafana/loki/blob/53416d6930e4d83515585c98b08866176fa8796c/pkg/validation/limits.go#L184

Reply:
Thank you!